scripts/ACMPermissions/PermissionFetcher.ts

```diff
@@ -0,0 +1,322 @@
+            event.topics[0] === eventFilter.topics[0][0] ? PermissionsEnum.Granted : PermissionsEnum.Revoked;
+
+          newEvents.push({
+            contractAddress: contractAddress,
+            functionSignature: functionSig,
+            account: account,
+            type: eventType,
+          });
+        } else {
+          this.addMissingRole(role, event.transactionHash);
+        }
+      });
+
+      return newEvents;
+    }
+
+    return events.map(event => {
+      const data = event.data;
+      const { account, contractAddress, functionSignature } = this.decodeLogs(data);
+      const eventType =
+        event.topics[0] === eventFilter.topics[0][0] ? PermissionsEnum.Granted : PermissionsEnum.Revoked;
+      return {
+        contractAddress: contractAddress,
+        functionSignature: functionSignature,
+        account: account,
+        type: eventType,
+      };
+    });
+  }
+
+  private getPermissionsJson(): { permissions: Permission[]; height: string } {
+    try {
+      if (!fs.existsSync(this.jsonFilePath)) {
+        return { permissions: [], height: startingBlockForACM[this.network].toString() };
+      }
+
+      const fileContent = fs.readFileSync(this.jsonFilePath, "utf8");
+
+      if (!fileContent.trim()) {
+        return { permissions: [], height: startingBlockForACM[this.network].toString() };
+      }
+      return JSON.parse(fileContent);
+    } catch (error) {
+      console.error(`Error parsing JSON file:`, error);
+      throw new Error("Error parsing JSON file");
+    }
+  }
+
+  private addMissingRole(role: string, transaction: string) {
+    let data: MissingRoleMap = {};
+
+    try {
+      if (fs.existsSync(this.missingRoleFile)) {
+        const fileContent = fs.readFileSync(this.missingRoleFile, "utf8").trim();
+        if (fileContent) {
+          try {
+            data = JSON.parse(fileContent);
+          } catch (error) {
+            console.error("Error parsing JSON:", error);
+            throw new Error("Error parsing JSON");
+          }
+        }
+      }
+
+      if (!data[role]) {
+        data[role] = { transactions: [] };
+      }
+
+      if (!data[role].transactions.includes(transaction)) {
+        data[role].transactions.push(transaction);
+      }
+
+      fs.writeFileSync(this.missingRoleFile, JSON.stringify(data, null, 2), "utf8");
+    } catch (error) {
+      console.error("Error adding missing role:", error);
+    }
+  }
+
+  private getLastBlockNumber(): number {
+    const fileContent = fs.readFileSync(this.jsonFilePath, "utf8");
+
+    const blockNumberRegex = /"height":\s*"(\d+)"/g;
+    const match = blockNumberRegex.exec(fileContent);
+    if (match) {
+      return parseInt(match[1], 10);
+    }
+    return startingBlockForACM[this.network];
+  }
+
+  private getHash(contractAddress: string, functionSignature: string): string {
+    const hash = ethers.utils.solidityKeccak256(["string", "string"], [contractAddress, functionSignature]);
```

Comment:
Why are we using here `["string", "string"]` instead of `["address", "string"]` as we are doing in line 170. Moreover, could we use `getHash` in line 170, instead of `solidityPack` + `keccak256`?